subprojects/tooling-api/src/main/java/org/gradle/tooling/events/task/TaskExecutionResult.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.tooling.events.task;
+
+import org.gradle.api.Incubating;
+import org.gradle.tooling.model.UnsupportedMethodException;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+/**
+ * Describes the result of a non-skipped task.
+ *
+ * @since 5.1
+ */
+@Incubating
+public interface TaskExecutionResult extends TaskOperationResult {
+
+    /**
+     * Returns whether this task was incremental.
+     *
+     * @return {@code true} if this task was incremental
+     * @throws UnsupportedMethodException For Gradle versions older than 5.1, where this method is not supported.
+     */
+    boolean isIncremental();
+
+    /**
+     * Returns the reasons why this task was executed.
+     *
+     * @return the reasons why this task was executed; an empty list indicates the task was up-to-date;
+     *         {@code null} that it failed before up-to-date checks had been performed.
```

Comment:
How about an `isUpToDate` instead of null here?

Reply:
You mean instead of _empty_, right? `TaskSuccessResult` already has an `isUpToDate()` property. For failed tasks that doesn't make sense. Alternatively, I could move both methods to `TaskSuccessResult` and `TaskFailureResult` and make it not nullable in the former. WDYT?

Reply:
No I meant avoiding a `null` List, but realize how this doesn't work out. Nevermind :)